eml/components/login/ResetPassword.js

```diff
@@ -14,59 +17,193 @@ import Text from '../general/Text';
  * - onModalClose: Function to do when modal closes
  */
 export default function ResetPassword(props) {
+  const emailAlertMessage = "Não existe nenhum usuário com este email!";
+  const [email, setEmail] = useState("");
+  const [token, setToken] = useState("");
   const [emailSent, setEmailSent] = useState(false);
   const [codeEntered, setCodeEntered] = useState(false);
+  const [passwordResetAlert, setPasswordResetAlert] = useState("");
+  const [tokenAlert, setTokenAlert] = useState("");
+
+  useEffect(() => {
+    if (email === '') {
+      setPasswordResetAlert('');
+      return;
+    }
+
+    const validationError = validateEmail(email);
+    setPasswordResetAlert(validationError);
+  }, [email]);
+
+  /**
+   * Function to send mail to user with code to reset password
+   * @param {*} email 
+   */
+
+  async function sendEmail(email) {
```

Comment:
We really need to add something to show the user, that the app is working (and disable the button) while this is being done. It takes a little time and no feedback is given to the user in the meanwhile. This would just prompt the user to keep clicking.